flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/table/descriptors/ElasticsearchValidator.java

```diff
@@ -87,7 +93,12 @@ private void validateHosts(DescriptorProperties properties) {
 		hostsValidators.put(CONNECTOR_HOSTS_HOSTNAME, (key) -> properties.validateString(key, false, 1));
 		hostsValidators.put(CONNECTOR_HOSTS_PORT, (key) -> properties.validateInt(key, false, 0, 65535));
 		hostsValidators.put(CONNECTOR_HOSTS_PROTOCOL, (key) -> properties.validateString(key, false, 1));
-		properties.validateFixedIndexedProperties(CONNECTOR_HOSTS, false, hostsValidators);
+
+		if (properties.containsKey(CONNECTOR_HOSTS)) {
+			validateAndGetHostsStr(properties);
```

Comment:
Using Validators too?

Reply:
we  need to  parse  hostsStr before validation when use new properties like :
`'connector.specific-offsets'='partition:0,offset:42;partition:1,offset:300'`
and the parse logic only used in ElasticSearch, so I add this function here

Reply:
Code can be reused, but validator can still go?

Reply:
we'd better parse and validate at the same time because `DescriptorProperties` do not support `Consumer` that can validate `String` value  like `partition:0,offset:42;partition:1,offset:300`